server/schedulers/hot_region.go

```diff
@@ -627,28 +488,50 @@ func (bs *balanceSolver) tryAddPendingInfluence() bool {
 	if bs.best == nil || len(bs.ops) == 0 {
 		return false
 	}
+	if bs.best.srcDetail.Info.IsTiFlash != bs.best.dstDetail.Info.IsTiFlash {
```

Comment:
It looks too late here. Should avoid this path earlier?

Reply:
It only occurs under specific wrong placement rules.